app/scripts/components/ipfs.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+
+const host = (process.env.NODE_ENV !== 'production') ? 'localhost' : window.location.hostname
+const port = (process.env.NODE_ENV !== 'production') ? '5001' : (window.location.port || 80)
+const ipfs = require('ipfs-api')(host, port)
+
+let version = ''
+let gateway = '//127.0.0.1:8080'
+
+export const withIpfs = (Component) => {
+  return React.createClass({
+    displayName: `withIpfs(${Component.displayName || Component.name || 'Component'})`,
+
+    getInitialState () {
+      return {version, gateway}
+    },
+
+    componentDidMount () {
+      ipfs.version((err, res) => {
+        if (err) return console.error(err)
```

Comment:
would be good to propagte these errors to the toastr state, rather than just logging them